pyexchange/uniswapv2_analytics.py

```diff
@@ -33,57 +34,100 @@
 class UniswapTrade(Trade):
 
     @staticmethod
-    def from_message(trade: dict, pair: str, token_a: Token, token_b: Token) -> Trade:
+    def from_our_trades_message(trade: dict, pair: str, base_token: Token, our_liquidity_balance: Wad, previous_base_token_reserves: Wad) -> Trade:
         """
         Assumptions:
             Prices are denominated in quote token
-            Aounts are denominated in the base token
+            Amounts are denominated in the base token
 
             We are always long the base token, and short quote
             if the amount of TokenA is 0, then we are essentially buying additional base token in exchange for quote token
 
+            If the base token reserves increase compared to the last block, then the trade is a buy
+
         Uniswap pairs can occasionally differ in base-quote terms from our expectations and price feeds.
         This requires a check to ensure that we are properly comparing asset pairs.
        """
         assert (isinstance(trade, dict))
         assert (isinstance(pair, str))
-        assert (isinstance(token_a, Token))
-        assert (isinstance(token_b, Token))
+        assert (isinstance(base_token, Token))
+        assert (isinstance(our_liquidity_balance, Wad))
+        assert (isinstance(previous_base_token_reserves, Wad))
 
-        if trade['pair']['token0']['id'] == token_a.address.address:
-            swap_price = Wad.from_number(trade['pair']['token1Price'])
+        our_pool_share = our_liquidity_balance / Wad.from_number(trade['pair']['totalSupply'])
```

Comment:
What happens if this is called when we have no liquidity in the pool?

Reply:
> If I understand correctly, this will synthesize a single trade for each hour liquidity was removed from the pool, based upon our supply of the token in that pool. As such, a consumer would be expected to query this once per hour. Is my understanding correct?

Yes, that is our current plan. This would be enforced by setting the `refresh-frequency` in `sync-trades`.

Reply:
If a user has no liquidity in the pool than it will return `[]`, based on the logic on line 277. If it would be better, I can also add an assertion to `from_our_trades_message` that will ensure the incoming `our_liquidity_balance` is greater than 0.